### DISCUSSION TITLE
Submit useful scripts here!

### DISCUSSION THREAD

**Discussion by ISSOtm** [gbdev/rgbds#1428] [Discussions]
:bulb: **Users can submit scripts they would like added to our [`contrib` folder](https://github.com/gbdev/rgbds/tree/master/contrib). The only requirement is that the script must be relevant to RGBDS, and useful to a large amount of our users. (Submit even if not sure, we're open to discussion!)**

---

Would it be relevant and useful to provide (and/or distribute) [this script](https://gist.github.com/ISSOtm/7d6c7409fdbd0f7a8e9d930b3eef3267) with RGBDS? Could be useful to us at least to help debug the CI (when `.bin` files don't match).

**Comment by mid-kid** [comment]
Sounds like something that belongs in an `extras/` or `contrib/` directory, as it's a bit too specific compared to the general use of this assembler.

**Comment by ISSOtm** [comment]
Thoughts, @AntonioND? New folder, separate repo, don't care at all?

**Comment by mid-kid** [comment]
If you start up a `contrib/` directory I recommend looking for similar scripts you may be able to add. https://github.com/pret/pokecrystal/blob/master/tools/free_space.awk comes to mind, for example, but maybe there's other simple tools that fit the bill (maybe my `unusedsymbols.py`? It's a bit on the bigger side though).

Of course, being a `contrib/` directory, it's not up to you to maintain such scripts, and you can drop them if they don't work anymore and nobody wants to fix them. It's more of an "interesting little scripts that might be useful to someone" thing. Nobody expects all of it to always work without modification. Kinda like awesome-gbdev in a way, except they probably don't want tiny rgbds-specific scripts.

**Comment by AntonioND** [comment]
If it can be of some use for the CI, I guess it can be in the repository. And yeah, it would make to have another folder like contrib like @mid-kid says.

**Comment by ISSOtm** [comment]
I added the script in b16ec83; as for adding other scripts, I would prefer if they had an explicit license before including them.
